### ISSUE TITLE
Add-on responsive to Firefox for Android

### ISSUE THREAD

**Issue by jhonatasrm** [mozillabrasil/sumo_live_helper#74]
I saw some users of the Add-on talking about use him on the Firefox for Android.

**Actual Behavior**
The Add-on works very well with Firefox for Android (that was not the focus) and some people are using on the Android device, but at the moment it's not responsive enough for a good experience.

**Expected Behavior**
The Add-on need present a good responsivity with Firefox for Android to promote a better experience possible.

P.S. We are currently working on a new design for the Add-on, but I think we can implement this responsivity before the new design comes out.

**Comment by WesleyBranton** [comment]
This is currently blocked by issue #58.

**Comment by WesleyBranton** [comment]
The add-on will need code to detect if it's running on a mobile device and then it will need to use that code to hide the sidebar button on mobile, since you obviously can't run a sidebar on mobile.

**Comment by WesleyBranton** [comment]
Here are a few things that I found wrong with the add-on on mobile:
- [ ] Preferences page needs to mobile-friendly (screen margins & larger font)
- [ ] Remove the add-on title from the header (in portrait)
- [ ] Question list not taking the full width (in landscape) (related to #108?)
- [ ] Loading bar isn't removed
- [ ] Hide sidebar button on mobile
- [ ] Disable sidebar setting on mobile

These would need to be fixed before we can mark the extension as an Android-compatible Firefox extension. If you want to see how it looks on mobile, you can use [Responsive Design Mode in Firefox](https://developer.mozilla.org/en-US/docs/Tools/Responsive_Design_Mode) if you view the `popup.html` file.

Currently, the differences between the popup for the sidebar vs the browser action are handled by adding a URL query (`?view=popup` or `?view=sidebar`) to the page that the `background.js` file opens. Then, the popup file adds a class value to the body of the popup when it first loads, depending on the value of the query. That's then used to hide certain UI elements (for example, the sidebar open button).

We could use the same process to handle the mobile popup page, by having the `background.js` file detect if the user is on mobile and handling the popups appropriately.

I'm not sure how urgent this is to implement, since I'm not sure how many contributors answer questions using a mobile device. When we are ready to start on this, I think we should create a new issue for each thing that needs changing so that we can assign people to the issues and we don't accidentally all work on the same issues. Of course, minor ones (like hiding elements) can be merged into a single issue or simply left here.

**Comment by dannycolin** [comment]
> Question list not taking the full width (in landscape)

It isn't related to the wrapping problem. This is due to the `max-width: 500px` (line 81 in css/app.css) to limit the popup size on desktop.

**Comment by WesleyBranton** [comment]
> > Question list not taking the full width (in landscape)
> 
> It isn't related to the wrapping problem. This is due to the `max-width: 500px` (line 81 in css/app.css) to limit the popup size on desktop.

Ok. Good to know. So it can easily be overridden with CSS then. That's good.

I swear there used to be a way to limit the popup size right from the manifest, but I guess not.

**Comment by WesleyBranton** [comment]
I'm think that maybe, for mobile, we have touch-friendly buttons at the top instead of the regular header. Just have the bar the same height as number, but the three buttons spread out in thirds across the top of the bar.

**Comment by dannycolin** [comment]
> I'm think that maybe, for mobile, we have touch-friendly buttons at the top instead of the regular header. Just have the bar the same height as number, but the three buttons spread out in thirds across the top of the bar.

Sounds good to me. I trust you on that one 'cause it's been a while since I used FF for Android and I don't have a device/emulator installed to see the end result with the mobile UI. ;)

**Comment by WesleyBranton** [comment]
The responsive design mode in Firefox is builtin and it will basically allow you see what it looks like on mobile.

**Comment by dannycolin** [comment]
> The responsive design mode in Firefox is builtin and it will basically allow you see what it looks like on mobile.

Sure but I was more thinking about not being able to see how well it integrate with the Firefox UI on that platform.
